src/snowflake/connector/telemetry_oob.py

```diff
@@ -382,6 +383,27 @@ def log_http_request_error(self,
             # Do nothing on exception, just log
             logger.debug("Failed to log HTTP request error", exc_info=True)
 
+    def log_general_exception(self,
+                              event_name: str,
+                              telemetry_data: dict,
+                              tags: Optional[dict] = None,
+                              urgent: Optional[bool] = False):
+        """Sends any type of exception through OOB telemetry."""
```

Comment:
Assuming from the comment above """Send telemetry data by in-band telemetry if it is enabled, otherwise send through out-of-band telemetry."""
This will be send if in-band is disabled. Can you be a little more explicit about it? What does it mean for in-band telemetry to be enabled / disabled. Is there a separate connection parameter / backend parameter that flips the switch on inband telemetry. It might be helpful for someone who doesn't have context and looking at this for the first time.

Reply:
Ah yes. So the intent is that if we hit an exception during connection setup we want telemetry on that exception which is where the OOB telemetry framework comes in. I just updated the logic so that we only send through OOB telemetry if the connection object is None. This way if the connection object exists and in-band telemetry is disabled we don't try to send it through OOB telemetry